src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -1009,6 +1009,12 @@ void loop_inst::set_memory_in_body_network(cldnn::network::ptr body_network,
                         "impl_params layout size(", impl_layout.to_short_string(),
                         ") should not exceed memory size(", updated_mem->get_layout().to_short_string(), ")");
         // Set need_to_check_memory_to_set to false to set output memory even if the input node has static shape,
+        if (impl_layout.bytes_count() < updated_mem->get_layout().bytes_count()) {
```

Comment:
I think this check condition creates redundant memory copies in cases where there is no memory conflict.

Reply:
@timxu826 From the description, you seem to resolve memory conflicts issue? 
If so, is this enough? 
Shouldn't we prevent memory reuse for the problematic case?